src/sdk.js

```diff
@@ -166,25 +166,34 @@ class Smartcar {
   /**
    * Calculate popup window size and position based on current window settings.
    *
+   * @param {Object} options - the postion and dimention setting of the popup window
+   * @param {String} [options.top] - the top property of
+   * [window features](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Window_features)
+   * @param {String} [options.left] - the left property of
+   * [window features](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Window_features)
+   * @param {String} [options.width] - the width property of
+   * [window features](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Window_features)
+   * @param {String} [options.height] - the height property of
+   * [window features](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Window_features)
    * @private
    * @return {String} a string of window settings
    */
-  static _getWindowOptions() {
+  static _getWindowOptions({top, left, width, height}) {
     // Sets default popup window size as percentage of screen size
     // Note that this only applies to desktop browsers
     const windowSettings = {
       width: window.screen.width * 0.3,
       height: window.screen.height * 0.75,
     };
 
-    const width = (window.outerWidth - windowSettings.width) / 2;
-    const height = (window.outerHeight - windowSettings.height) / 8;
+    const widthOffset = (window.outerWidth - windowSettings.width) / 2;
+    const heightOffset = (window.outerHeight - windowSettings.height) / 8;
 
     let options = '';
-    options += `top=${window.screenY + height},`;
-    options += `left=${window.screenX + width},`;
-    options += `width=${windowSettings.width},`;
-    options += `height=${windowSettings.height},`;
+    options += `top=${top || window.screenY + heightOffset},`;
```

Comment:
Do we want to allow values of `0` for top/left/width/height? Cause `0` would always be considered falsy otherwise.

Reply:
Thought that these configs should be strings only, added more handlings for number and invalid strings.